Sally7/RequestExecutor/ConcurrentRequestExecutor.cs

```diff
@@ -209,13 +210,17 @@ public JobPool(int maxNumberOfConcurrentRequests)
                 }
             }
 
-            public void Dispose() => _jobIdPool.Writer.Complete();
+            public void Dispose()
+            {
+                Volatile.Write(ref _disposed, true);
+                _jobIdPool.Writer.Complete();
+            }
 
             public ValueTask<int> RentJobIdAsync(CancellationToken cancellationToken) => _jobIdPool.Reader.ReadAsync(cancellationToken);
 
             public void ReturnJobId(int jobId)
             {
-                if (!_jobIdPool.Writer.TryWrite(jobId))
+                if (!_jobIdPool.Writer.TryWrite(jobId) && !Volatile.Read(ref _disposed))
```

Comment:
Is there a reason for the ordering? Wouldn't you want the `!Volatile.Read(ref _disposed)` checked first, and the TryWrite only executed if it isn't yet disposed, taking care of short-circuiting `&&`.

Or is there a reason for always executing the writing to the jobIdPool, independent of it being disposed or not?

Reply:
TryWrite is called more often than Dispose so I think that order is fine, meaning that the common path comes first.
If TryWrite fails, then it's checked if disposed or not.

But TBH I don't think that any order here actually matters. The arguing of @scamille is also fine.
Both yield the same result, and perf-wise there's no difference (at least with TCP connection to PLC in mind). And I have no problem if the order is flipped.